crypto/src/key/mod.rs

```diff
@@ -57,6 +70,16 @@ pub struct PublicKey {
     pub_key: PublicKeyHolder,
 }
 
+#[derive(Debug, PartialEq, Eq, Clone, Decode, Encode)]
```

Comment:
Please move all these new things into an `extended.rs` module. You're also free to move everything else and make this file just for `mod` and `use` statements. We try everything we can to avoid clutter in code.